packages/cache/src/internal/cacheHttpClient.ts

```diff
@@ -113,6 +114,21 @@ export async function getCacheEntry(
   const cacheResult = response.result
   const cacheDownloadUrl = cacheResult?.archiveLocation
   if (!cacheDownloadUrl) {
+    // List cache for primary key only if cache miss occurs
+    const resource = `caches?key=${encodeURIComponent(keys[0])}`
+    const response = await httpClient.getJson<ArtifactCacheList>(getCacheApiUrl(resource))
+    if(response.statusCode === 204) {
+      const cacheListResult = response.result
+      const totalCount = cacheListResult?.totalCount
+      if(totalCount && totalCount > 0) {
+        core.info(`Cache miss occurred on the cache key '${keys[0]}' and version '${version} but there is ${totalCount} existing version of the cache for this key. More info on versioning can be found here: https://github.com/actions/cache#cache-version`)
+        core.debug(`Other versions are as follows:`)
```

Comment:
I don't think count is much valuable info here and can be removed.

```suggestion
        core.info(`No matching cache found for cache key '${keys[0]}' with version '${version}. More info on cache matching can be found here: https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key.`)
        core.debug(`Candidate caches which were tried:`)
```

Reply:
Makes sense. I added cache version docs to tell that version is the reason of mismatch. Candidate caches were not tried. They are other existing caches with same key but different versions

Reply:
Removed "cache miss"